fs/compress.py

```diff
@@ -76,7 +77,13 @@ def write_zip(src_fs,
                 )
             zip_info = zipfile.ZipInfo(zip_name, zip_time)
 
+            try:
+                zip_info.external_attr = info.permissions.mode << 16
+            except MissingInfoNamespace:
+                zip_info.external_attr = 0o600
```

Comment:
Does this not have to be shifted left by 16?

Reply:
I think it would be even better to leave it as 0, which is the actual default anyway.